src/os/src/detail/osTime.c

```diff
@@ -614,36 +614,25 @@ int64_t taosTimeTruncate(int64_t timeStamp, const SInterval* pInterval, int32_t
 
     if (pInterval->slidingUnit == 'y') {
       tm.tm_mon = 0;
-      tm.tm_year = (int)(tm.tm_year / pInterval->sliding * pInterval->sliding);
+      tm.tm_year -= (int)(tm.tm_year%pInterval->sliding);
     } else {
       int mon = tm.tm_year * 12 + tm.tm_mon;
-      mon = (int)(mon / pInterval->sliding * pInterval->sliding);
+      mon -= (int)(mon % pInterval->sliding);
```

Comment:
The effect of the new statement keeps the same as the old code. It is changed to unify the method of truncating aligned local time intervals.